node/heartbeat/monitor.go

```diff
@@ -237,9 +244,9 @@ func (m *Monitor) SetAppStatusHandler(ash core.AppStatusHandler) error {
 		return ErrNilAppStatusHandler
 	}
 
-	m.mutAppStatusHandler.Lock()
+	m.mutAppStatusHandler.RLock()
```

Comment:
Should revert to Lock-Unlock as you set the pointer

Reply:
fixed